src/index.js

```diff
@@ -1,15 +1,24 @@
 var mime = require('mime-types');
-var fs = require('fs');
+var glob = require('glob');
 
-var preCompressedFiles = [];
+var preCompressedFiles = {};
 var extensionRegex = /(\.(br|gz))$/;
 
 module.exports = function preCompressAssets(urlRegexp, publicPath) {
   var fileRegex = new RegExp(urlRegexp.source + extensionRegex.source);
-  fs.readdir(publicPath, function(err, files){
-    files.forEach(function(file) {
-      if (fileRegex.test(file)) {
-        preCompressedFiles.push('/' + file);
+  var indexRegex = new RegExp('index.html' + extensionRegex.source);
+
+  glob(publicPath + '/**/*', function (err, filePaths) {
+    filePaths.forEach(function(filePath) {
+      if (fileRegex.test(filePath)) {
+        var reducedFilePath = filePath.replace(publicPath, '');
+        preCompressedFiles[reducedFilePath] = true;
```

Comment:
When use in [hexo-server](https://github.com/hexojs/hexo-server), I need to prepend a slash to `reducedFilePath`. Either through `preCompressedFiles['/' + reducedFilePath]` or `filePath.replace(publicPath, '/')`.

In my use case, I don't have to `path.join(__dirname, 'public')`, instead it's just `'public'`. Perhaps this explains the lack of slash?

I wonder if `reducedFilePath` has starting slash in your case?